src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -329,20 +311,61 @@ private async Task IndexAddressesContinuouslyAsync()
                 }
 
                 this.IndexerTip = nextHeader;
+            }
+
+            this.SaveAll();
+        }
 
-                lock (this.lockObject)
+        private void RewindAndSave(ChainedHeader rewindToHeader)
+        {
+            lock (this.lockObject)
+            {
+                // The cache doesn't really lend itself to handling a reorg very well.
+                // Therefore, we leverage LiteDb's indexing capabilities to tell us
+                // which records are for the affected blocks.
+                // TODO: May also be efficient to run ProcessBlocks with inverted deposit flags instead, depending on size of reorg
```

Comment:
I think this TODO is not really applicable any more with the changes to the rewind logic.